libs/custom-paper-handler/src/driver/constants.ts

```diff
@@ -17,6 +17,31 @@ export const START_OF_PACKET: Uint8 = 0x02;
 export const OK_CONTINUE: Uint8 = 0x00;
 export const OK_NO_MORE_DATA: Uint8 = 0xff;
 
+export type PrintingDensity = '-25%' | '-12.5%' | 'default' | '+12.5%' | '+25%';
+export type PrintingSpeed = 'slow' | 'normal' | 'fast';
+export const PRINTING_SPEED_CODES: Record<PrintingSpeed, Uint8> = {
+  slow: 0,
+  normal: 1,
+  fast: 2,
+};
+
+export const PRINTING_DENSITY_CODES: Record<PrintingDensity, Uint8> = {
+  '-25%': 0x02,
+  '-12.5%': 0x03,
+  default: 0x04,
+  '+12.5%': 0x05,
+  '+25%': 0x06,
+};
+// Exhaustive list of scan types whose data blocks will be 8 bpp.
+// Manual reference: "Start scan ticket" 0x1C 0x53 0x50 0x53
+export enum ScanTypes8BitsPerPixel {
+  NA = 0x00,
+  RED = 0x01,
+  GREEN = 0x02,
+  BLUE = 0x03,
+  ULTRAVIOLET = 0x04,
+  GRAY = 0x05,
+}
```

Comment:
I find this name a little misleading because the actual return value from "Start scan ticket" does _not_ need to represent an 8bpp value. You're only encoding those values that _do_ represent them and then using presence in the enum to imply 8bpp. I think this might be clearer, if more verbose:

```suggestion
// Exhaustive list of scan types.
// Manual reference: "Start scan ticket" 0x1C 0x53 0x50 0x53
export enum ScanTypes {
  NA = 0x00,
  /**
   * Red light scan (8bpp)
   */
  RED = 0x01,
  /**
   * Green light scan (8bpp)
   */
  GREEN = 0x02,
  /**
   * Blue light scan (8bpp)
   */
  BLUE = 0x03,
  /**
   * Ultraviolet light scan (8bpp)
   */
  ULTRAVIOLET = 0x04,
  /**
   * White light scan (8bpp)
   */
  GRAY = 0x05,
  /**
   * first line RED, second line GREEN, third line BLUE (24bpp)
   */
  RGB_RAW = 0x06,
  /**
   * first line RED, second line GREEN, third line BLUE, fourth line UV (32bpp)
   */
  RGBU_RAW = 0x07,
  /**
   * Black and white with red light (1bpp)
   */
  BW_RED = 0x08,
  /**
   * Black and white with green light (1bpp)
   */
  BW_GREEN = 0x09,
  /**
   * Black and white with blue light (1bpp)
   */
  BW_BLUE = 0x0a,
  /**
   * Black and white with ultraviolet light (1bpp)
   */
  BW_ULTRAVIOLET = 0x0b,
  /**
   * Black and white with white light (1bpp)
   */
  BW = 0x0c,
}

export function getBitsPerPixelForScanType(scanType: ScanTypes): number {
  switch (scanType) {
    case ScanTypes.NA:
      return 0;
    case ScanTypes.RED:
    case ScanTypes.GREEN:
    case ScanTypes.BLUE:
    case ScanTypes.ULTRAVIOLET:
    case ScanTypes.GRAY:
    case ScanTypes.BW_RED:
    case ScanTypes.BW_GREEN:
    case ScanTypes.BW_BLUE:
    case ScanTypes.BW_ULTRAVIOLET:
    case ScanTypes.BW:
      return 8;
    case ScanTypes.RGB_RAW:
      return 24;
    case ScanTypes.RGBU_RAW:
      return 32;
    default:
      throwIllegalValue(scanType);
  }
}
```

Reply:
I agree with your take, thanks for the suggestion. I'll merge this in.

Reply:
https://github.com/votingworks/vxsuite/pull/3756/commits/51a0803a462c3a7eb61208d3925e6a07f4989f8f